### ISSUE TITLE
Checking Express event content in Configuration/DataProcessing 

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#22933]
While checking #22796 to verify the changes in the event content, I tried to check not only the usual RelVals (in wf 10824 the new totemTiming digi collection is added to the AODSIM event) but also what would happen in express. 

To do this, after having merged #22796 in my work area and added Configuration/DataProcessing, I produced in the test area of this package the express configuration with the command

python RunExpressProcessing.py --scenario ppEra_Run2_2018 --global-tag GLOBALTAG --lfn /store/whatever --fevt --dqmio --alcarecos=TkAlMinBias+SiStripCalZeroBias

and then expanded it with edmConfigDump.

The various process.RecoCTPPSFEVT , AOD etc. PSets for outputCommands are visible, but the event content in the OutputModule is not containing them, while this is not the case when I expand the configuration for step3 of 10824.

Of course the input file, GT above are dummy and I do not expect the configuration to run, but I would expect its content is correct from the python point of view.

What is wrong with this test? In a similar way also muonGEMDigis are absent from the OutputModule.

**Comment by fabiocos** [comment]
@slava77 @perrotta here is the follow-up of our direct discussion

**Comment by slava77** [comment]
The problem is apparently with the eras/process parsing for the EventContent.
Python that's generated with cmsDriver is OK, but the creation of ConfigBuilder and then a pickle as done in the data processing fails to pick up era-based modifications to the event content.

this problem showed up in 10_0_1 and was not present in 10_0_0.

I tried 10_0_0+ #21953 and see the problem is due to this PR.
Basically, the Configuration.EventContent.EventContent_cff at that point is loaded before the eras exist.
This breaks the T0 configuration.

@fabiocos 
good catch.
We should add some better testing in the DataProcessing and perhaps just add a workflow that runs tests including comparisons using this machinery.

**Comment by fabiocos** [comment]
@slava77 thank you for the investigation, this needs to be fixed urgently

**Comment by slava77** [comment]
Actually, if the relval submission relies on the same DMWM machinery, we should be missing all this data in centrally submitted relvals as well.

This refers to all centrally submitted workflows with the event content modified with eras (HI on pp; 2016 and later for CTPPS and GEM data; probably also re-miniAOD workflows for more commonly used objects)

**Comment by fabiocos** [comment]
@slava77 but this is not visible in tests because they are not using that interface...

**Comment by slava77** [comment]
On 4/11/18 8:35 AM, Fabio Cossutti wrote:
> @slava77 <https://github.com/slava77> but this is not visible in tests
> because they are not using that interface...
> 

right
what I'm suggesting is to add such tests using the python object based
configuration or perhaps even the full path via DMWM code so that we
actually test what we run.

**Comment by fabiocos** [comment]
@slava77 yes, definitely we need to have a better test of this machinery

@arizzi @franzoni FYI 

I am indeed concerned about the potential effect of this on the work done so far for 94X reminiAOD

**Comment by Dr15Jones** [comment]
Seems like `ConfigBuilder` should _turn on_ the requested `Modifier`s at the very beginning, before doing any imports.

**Comment by slava77** [comment]
On 4/11/18 8:44 AM, Chris Jones wrote:
> Seems like |ConfigBuilder| should /turn on/ the requested |Modifier|s at
> the very beginning, before doing any imports.
> 

In this case the code is loaded before the ConfigBuilder is loaded.
Can the ConfigBuilder figure it out?

**Comment by arizzi** [comment]
I'm not sure I understand why/how this fails (mainly because I do not know how the Era modifier works.)

I do not think in reminiaod we have any era modifier on th event content if that's the worry (but I can be wrong) @gpetruc  ?

**Comment by fabiocos** [comment]
@slava77 as this problem is present also in 94X, although it should affect only T0 (see the thread https://hypernews.cern.ch/HyperNews/CMS/get/wmDevelopment/793/1.html ) it is advisable to fix it also there

**Comment by slava77** [comment]
> @slava77 as this problem is present also in 94X, although it should affect only T0 (see the thread https://hypernews.cern.ch/HyperNews/CMS/get/wmDevelopment/793/1.html ) it is advisable to fix it also there

done in #22940
